flink-libraries/flink-table/src/main/scala/org/apache/flink/table/formats/TableFormatFactoryService.scala

```diff
@@ -0,0 +1,245 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.formats
+
+import java.util.{ServiceConfigurationError, ServiceLoader, Map => JMap}
+
+import org.apache.flink.table.api._
+import org.apache.flink.table.descriptors.FormatDescriptorValidator.FORMAT_PROPERTY_VERSION
+import org.apache.flink.table.descriptors._
+import org.apache.flink.table.util.Logging
+
+import _root_.scala.collection.JavaConverters._
+import _root_.scala.collection.mutable
+
+/**
+  * Service provider interface for finding a suitable [[TableFormatFactory]] for the
+  * given properties.
+  */
+object TableFormatFactoryService extends Logging {
+
+  private lazy val defaultLoader = ServiceLoader.load(classOf[TableFormatFactory[_]])
+
+  /**
+    * Finds a table format factory of the given class and creates configured instances from the
+    * given descriptor.
+    *
+    * @param factoryClass desired format factory
+    * @param descriptor descriptor that describes the format
+    * @tparam T factory class type
+    * @return configured instance from factory
+    */
+  def find[T](factoryClass: Class[T], descriptor: Descriptor): T = {
+    find(factoryClass, descriptor, null)
+  }
+
+  /**
+    * Finds a table format factory of the given class and creates configured instances from the
+    * given descriptor and classloader.
+    *
+    * @param factoryClass desired format factory
+    * @param descriptor descriptor that describes the format
+    * @param classLoader classloader for service loading
+    * @tparam T factory class type
+    * @return configured instance from factory
+    */
+  def find[T](
+      factoryClass: Class[T],
+      descriptor: Descriptor,
+      classLoader: ClassLoader)
+    : T = {
+
+    val properties = new DescriptorProperties()
+    descriptor.addProperties(properties)
+    find(factoryClass, properties.asMap, classLoader)
+  }
+
+  /**
+    * Finds a table format factory of the given class and creates configured instances from the
+    * given property map.
+    *
+    * @param factoryClass desired format factory
+    * @param propertyMap properties that describes the format
+    * @tparam T factory class type
+    * @return configured instance from factory
+    */
+  def find[T](factoryClass: Class[T], propertyMap: JMap[String, String]): T = {
+    find(factoryClass, propertyMap, null)
+  }
+
+  /**
+    * Finds a table format factory of the given class and creates configured instances from the
+    * given property map and classloader.
+    *
+    * @param factoryClass desired format factory
+    * @param propertyMap properties that describes the format
+    * @param classLoader classloader for service loading
+    * @tparam T factory class type
+    * @return configured instance from factory
+    */
+  def find[T](
```

Comment:
split this method into smaller ones. Basically everywhere where you typed a comment 
```
//foo bar baz
some;
code;
block;
```
replace the comment with method call:
```
def fooBarBaz() {
  some;
  code;
  block;
}
```